File: backend/src/main/java/com/pofo/backend/domain/resume/education/entity/Education.java
package com.pofo.backend.domain.resume.education.entity;

import com.pofo.backend.common.jpa.entity.BaseTime;
import com.pofo.backend.domain.resume.resume.entity.Resume;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "educations")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@Builder(toBuilder = true)
public class Education extends BaseTime {

    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private String major;
    @Column(nullable = false)
    private LocalDate startDate;
    @Column(nullable = false)
    private LocalDate endDate;
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Status status;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "resume_id", nullable = false)
    private Resume resume;

    public enum Status {
        EXPECTED, // 예정
        GRADUATED, // 졸업
        ENROLLED, // 재학
        REST; // 휴학
    }
}

File: backend/src/main/java/com/pofo/backend/domain/resume/education/entity/Education.kt
package com.pofo.backend.domain.resume.education.entity

import com.pofo.backend.common.jpa.entity.BaseTime
import com.pofo.backend.domain.resume.resume.entity.Resume
import jakarta.persistence.*
import lombok.Getter
import java.time.LocalDate

@Entity
@Table(name = "educations")
@Getter
class Education(

    @Column(nullable = false)
    val name: String,

    @Column(nullable = false)
    val major: String,

    @Column(nullable = false)
    val startDate: LocalDate,

    @Column(nullable = false)
    val endDate: LocalDate,

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    val status: Status,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "resume_id", nullable = false)
    val resume: Resume

) : BaseTime() {

    enum class Status {
        EXPECTED,  // 예정
        GRADUATED,  // 졸업
        ENROLLED,  // 재학
        REST // 휴학
    }
}
